### PR TITLE
solution

### PR DESCRIPTION
**Pull request by maksym-kostetskyi**
https://maksym-kostetskyi.github.io/react_list-of-goods-js/

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -14,45 +19,162 @@ export const goodsFromServer = [
...
+  if (reversed) {
+    return reversedGoods;
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
```
Comment (Viktor-Kost) [inline src/App.jsx:64]:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

File: src/App.jsx [hunk 2]
```diff
@@ -14,45 +19,162 @@ export const goodsFromServer = [
...
+  const [nameClicked, setNameClicked] = useState(false);
+  const [resetState, setResetState] = useState(false);
+  const [lengthClicked, setLengthClicked] = useState(false);
```
Comment (Viktor-Kost) [inline src/App.jsx:67]: you should use only one state (sortField) for all of these cases

File: src/App.jsx [hunk 3]
```diff
@@ -14,45 +19,162 @@ export const goodsFromServer = [
...
+  const visibleGoods
+  = getPreparedGoods(goodsFromServer, { sortField, reversed });
+
+  function clickName() {
```
Comment (Viktor-Kost) [inline src/App.jsx:71]: depends on what you click you will just set a correct sortField. 
So you don't need to have all of these handlers

File: src/App.jsx [hunk 4]
```diff
@@ -14,45 +19,162 @@ export const goodsFromServer = [
...
+        setResetState(false);
+      }
+    } else {
+      setReversed(true);
+      reverseButton.classList.remove('is-light');
+      setResetState(true);
+    }
+  }
+
+  function clickReset() {
```
Comment (Viktor-Kost) [inline src/App.jsx:117]: you need just to reset `isReversed` and `sortField`

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/App.jsx [hunk 5]
```diff
@@ -14,45 +19,140 @@ export const goodsFromServer = [
...
+          : (
+            <button
+              type="button"
+              id="reverse"
+              className="button is-warning is-light"
+              onClick={() => setIsReversed(!isReversed)}
+            >
+              Reverse
+            </button>
+          )}
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:137]:
```suggestion
            <button
              type="button"
              id="reverse"
              className={cn("button is-warning", { isLight: !isReversed })}
              onClick={() => setIsReversed(prevIsReversed => !prevIsReversed)}
            >
              Reverse
            </button>
```
 use classnames library for dynamic set classNames. no need repeat jsx for it.
 same for other buttons
 
better change isReversed from prev state

File: src/App.jsx [hunk 6]
```diff
@@ -14,45 +19,140 @@ export const goodsFromServer = [
...
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return reversedGoods;
+  }
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:59]:
```suggestion
    preparedGoods.sort((good1, good2) => {
      switch (sortField) {
        case SORT_FIELD_NAME:
          return good2.localeCompare(good1);

        case SORT_FIELD_LENGTH:
          return good2.length - good1.length;

        default:
          return 0;
      }
    });

  if (isReversed) {
    return preparedGoods.reverse();
  }
```
just reverse if it needs after sorting.

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Almost done.
Suggest some refactoring.

**Review by VitaliyBondarenko1982 — APPROVED** [review]
Well done

**Pull request closed without merging**